src/node.rs

```diff
@@ -680,4 +668,44 @@ mod test {
         node.entries_mut().push(KdlEntry::new_prop("x", 2));
         assert_eq!(&node["x"], &2.into())
     }
+
+    #[test]
+    fn insertion() {
+        let mut node = KdlNode::new("foo");
+        node.push("pos0");
+        node.insert("keyword", 6.0);
+        node.push("pos1");
+        assert_eq!(node.entries().len(), 3);
+
+        node.insert(0, "inserted0");
+        node.insert(1, "inserted1");
+        assert_eq!(node.entries().len(), 3);
+        assert_eq!(node[0], "inserted0".into());
+        assert_eq!(node[1], "inserted1".into());
+    }
+
+    #[test]
+    fn removal() {
+        let mut node = KdlNode::new("foo");
+        node.push("pos0");
+        node.insert("keyword", 6.0);
+        node.push("pos1");
+        assert_eq!(node.entries().len(), 3);
+
+        node.remove(1);
+        assert_eq!(node.entries().len(), 2, "index removal should succeed");
+        assert!(
+            node.get("keyword").is_some(),
+            "keyword property should not be removed by index removal"
+        );
+        node.remove(1);
+        assert_eq!(node.entries().len(), 2, "index removal should not succeed");
+        node.remove("not an existing keyword");
+        assert_eq!(node.entries().len(), 2, "key removal should not succeed");
+        node.remove("keyword");
+        assert_eq!(node.entries().len(), 1, "key removal should succeed");
+        node.remove(0);
+        assert_eq!(node.entries().len(), 0, "index removal should suceed");
+        node.remove(0); // should not panic
```

Comment:
maybe it should panic? That's what [`Vec::remove`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.remove) does, after all.

Reply:
Seems reasonable to me